tests/lib/Controller/Trait/ErrorTraitTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\Module\authoauth2\Tests\Controller\Trait;
+
+use DG\BypassFinals;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\InputBag;
+use SimpleSAML\Module\authoauth2\Controller\Traits\ErrorTrait;
+
+class ErrorTraitTest extends TestCase
+{
+    use ErrorTrait;
+
+    public function setUp(): void
+    {
+        BypassFinals::enable(bypassReadOnly: false);
+    }
+
+    /**
+     * @throws \PHPUnit\Framework\MockObject\Exception
+     */
+    public function testParseErrorWithNoError(): void
+    {
+        // Mocking Request object
+        $request = $this->createMock(Request::class);
+        $request->query = $this->createMock(InputBag::class);
+
+        // Stubbing the method has() and get() of InputBag
+        $request->query->method('has')
+            ->willReturnOnConsecutiveCalls(false, false);
+
+        // Test
+        $result = $this->parseError($request);
+        $this->assertEquals(['', ''], $result);
+    }
+
+    public function testParseErrorWithErrorAndDescription(): void
+    {
+        // Mocking Request object
+        $request = $this->createMock(Request::class);
```

Comment:
I think these tests aren't doing anything.  If I change the `ErrorTrait`  to use a different query param  (e.g. `error` to `error2` )the tests all still pass.  I think we  would be better if you didn't mock the request and just created one with `Request::create(`  that contains the actual query keys.

Reply:
Done